hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/transaction/lock/StorageBasedLockProvider.java

```diff
@@ -267,6 +268,18 @@ private synchronized void shutdown(boolean fromShutdownHook) {
     } catch (Exception e) {
       logger.error("Owner {}: Heartbeat manager failed to close.", ownerId, e);
     }
+    try {
+      this.auditService.ifPresent(auditService -> {
+        try {
+          auditService.close();
+        } catch (Exception e) {
+          logger.error("Owner {}: Audit service failed to close.", ownerId, e);
+        }
+      });
+      this.auditService = Option.empty();
+    } catch (Exception e) {
+      logger.error("Owner {}: Failed to close audit service.", ownerId, e);
+    }
```

Comment:
Should audit service also be closed if `fromShutdownHook=true`?

Reply:
no, if the shutdown hook is invoked we do not want to do anything extra. notice we don't close the storage lock client either

Reply:
Got it.